core/tests/protractor_utils/ExplorationEditorFeedbackTab.js

```diff
@@ -56,122 +56,123 @@ var ExplorationEditorFeedbackTab = function() {
+    var rows = await element.all(by.css(suggestionRowClassName));
+    var matchingSuggestionRows = [];
+    for (var i = 0; i < rows.length; i++) {
+      var subject = await rows[i].element(
+        by.css('.protractor-test-exploration-feedback-subject')).getText();
+      if (suggestionDescription.indexOf(subject) !== -1) {
+        matchingSuggestionRows.push(await rows[i]);
+      }
+    }
+    expect(await matchingSuggestionRows[0].isDisplayed()).toBe(true);
+    await matchingSuggestionRows[0].click();
+    expect(await viewSuggestionButton.isDisplayed()).toBe(true);
+    await viewSuggestionButton.click();
+    expect(await acceptSuggestionButton.isDisplayed()).toBe(true);
+    await suggestionCommitMessageInput.sendKeys('Commit message');
+    await acceptSuggestionButton.click();
+    await waitFor.invisibilityOf(
+      acceptSuggestionButton, 'Suggestion modal takes too long to disappear');
+    await waitFor.pageToFullyLoad();
   };
 
-  this.expectToHaveFeedbackThread = function() {
-    expect(feedbackTabRow.isPresent()).toBe(true);
+  this.expectToHaveFeedbackThread = async function() {
+    expect(await feedbackTabRow.isPresent()).toBe(true);
   };
 
-  this.getSuggestionThreads = function() {
+  this.getSuggestionThreads = async function() {
     var threads = [];
-    waitFor.visibilityOf(
+    await waitFor.visibilityOf(
       element.all(by.css(suggestionRowClassName)).first(),
       'No suggestion threads are visible');
-    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
-      rows.forEach(function() {
-        explorationFeedbackSubject.getText().then(function(subject) {
-          threads.push(subject);
-        });
-      });
-      return threads;
-    });
+    var rows = await element.all(by.css(suggestionRowClassName));
+    for (var i = 0; i < rows.length; i++) {
+      var subject = await explorationFeedbackSubject.getText();
+      threads.push(subject);
+    }
+    return threads;
   };
 
-  this.goBackToAllFeedbacks = function() {
-    feedbackBackButton.click();
+  this.goBackToAllFeedbacks = async function() {
+    await feedbackBackButton.click();
   };
 
-  this.readFeedbackMessages = function() {
+  this.readFeedbackMessages = async function() {
     var messages = [];
-    waitFor.visibilityOf(
+    await waitFor.visibilityOf(
       element.all(by.css(suggestionRowClassName)).first(),
       'No feedback messages are visible.');
-    return element.all(by.css(suggestionRowClassName)).then(function(rows) {
-      rows.forEach(function(row) {
-        row.click();
-        waitFor.visibilityOf(
-          explorationFeedback, 'Feedback message text is not visible');
-        explorationFeedback.getText().then(function(message) {
-          messages.push(message);
-        });
-        feedbackBackButton.click();
-      });
-      return messages;
-    });
+    var rows = await element.all(by.css(suggestionRowClassName));
```

Comment:
ditto: don't await `element.all` and use `.count()` and `.get()` instead of array operations below. Also elsewhere in this file